signature/src/signatures/sigstore.rs

```diff
@@ -0,0 +1,167 @@
+            let mut new_docker = HashMap::new();
+            new_docker.insert(ns_name.to_string(), ns_config.clone());
+            self.docker = Some(new_docker)
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+struct RegistryNamespace {
+    sigstore: String,
+}
+
+// loads sigstore configuration files(.yaml files) in specific dir.
+fn load_sigstore_config(dir: &str) -> Result<SigstoreConfig> {
+    let mut merged_config = SigstoreConfig::default();
+    let yaml_extension = OsStr::new("yaml");
+
+    for entry in fs::read_dir(dir)? {
+        let entry = entry?;
+        let path = entry.path();
+        if path.is_dir() || path.extension() != Some(yaml_extension) {
+            continue;
+        }
+        let path_str = path
+            .to_str()
+            .ok_or(anyhow!("Unknown error: path parsed failed."))?;
+        let config_yaml_string = fs::read_to_string(path_str)?;
+        let config = serde_yaml::from_str::<SigstoreConfig>(&config_yaml_string)?;
+
+        // The "default-docker" only allowed to be defined in one config file.
+        if config.default_docker.is_some() {
+            if merged_config.default_docker.is_some() {
+                return Err(anyhow!(
+                    "Error parsing sigstore config: \"default-docker\" defined repeatedly."
+                ));
+            }
+            merged_config.default_docker = config.default_docker;
+        }
+
+        // An image namespace is not allowed appear in two different configuration files.
+        if let Some(docker_config_map) = config.docker {
+            for (ns_name, ns_config) in docker_config_map.iter() {
+                if merged_config.is_config_entry_exit(ns_name) {
+                    return Err(anyhow!(format!(
+                        "Error parsing sigstore config: {} defined repeatedly.",
+                        &ns_name
+                    )));
+                }
+
+                merged_config.insert_a_docker_config_entry(ns_name, ns_config);
+            }
+        }
+    }
+
+    Ok(merged_config)
+}
+
+// returns an URL string (sigstore base url) configured in sigstore config for the image.
+pub fn get_sigstore_base_url(image_ref: &Reference) -> Result<String> {
+    let sigstore_config = load_sigstore_config(REGISTRY_CONFIG_DIR)?;
+
+    if let Some(docker_config_map) = sigstore_config.docker {
+        // Look for a full match.
+        let image_id = image_ref.whole();
+        // Look for a match of the possible parent namespaces.
+        let mut image_ns = image::get_image_namespaces(image_ref);
+
+        image_ns.insert(0, image_id);
+        for ns in image_ns.iter() {
+            if let Some(base_url) = docker_config_map.get(ns) {
+                return Ok(base_url.sigstore.to_string());
+            }
+        }
+    }
+
+    // Look for a default location
+    if let Some(default_config) = sigstore_config.default_docker {
+        return Ok(default_config.sigstore);
```

Comment:
Shouldn't you convert to string (i.e. `default_config.sigstore.to_string()`) here as well?